src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+
+export default function Icon ( props: {
+	/**
+	 * Fill color of the icon to be rendered. Can be any HTML-valid
+	 * color string, e.g. hexadecimal, RGBA, CSS color names, etc.
+	 * If supplied, this value takes precedence over `props.themeColor`.
+	 */
+	color?: string,
+
+	/**
+	 * Name of the icon to be rendered.
+	 */
+	name: (
+		'apple' |
```

Comment:
we could make the icons list a plain ol' object so you don't have to type the name of each object twice — typescript can infer a union type from the keys of the object. like:

```typescript
const svgIcons = {
	'apple': ( props ) => ( <svg {...props}>...</svg> ) ),
};
```

then the type of the `name` prop is simply `name: keyof typeof Icons` instead of the union of string names

and instead of `switch` you would just do

```typescript
const Icon = svgIcons[props.name];
return <Icon {...commonProps}/>;
```

Reply:
ah nice idea

Reply:
Hmm this doesn't seem to work, or eslint doesn't like it at any rate. Could you point me to anywhere you've seen this used?
![Screen Shot 2021-03-01 at 3 41 37 PM](https://user-images.githubusercontent.com/591884/109557724-7883de80-7aa6-11eb-86fc-9c5079d88ca5.png)